deployer/src/handlers/error.rs

```diff
@@ -0,0 +1,42 @@
+use std::error::Error as StdError;
+
+use axum::http::{header, HeaderValue, StatusCode};
+use axum::response::{IntoResponse, Response};
+use axum::Json;
+
+use serde::{ser::SerializeMap, Serialize};
+use serde_json::json;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Persistence failure: {0}")]
+    Persistence(#[from] crate::persistence::PersistenceError),
+}
+
+impl Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("type", &format!("{:?}", self))?;
```

Comment:
I know you moved this over from somewhere else, so not really part of this PR, but I couldn't help but notice:

Why `{:?}` and why `self`? Won't this inline the entire debug disposition of the error as its "type"? Would it be better to have a `match` and display the variant name?

Reply:
Yeah, this will have the entire debug representation as the "type". With an example error the serialized output will be: `{"message":{"type":"Persistence(Database(PoolTimedOut))","msg":"Database error: pool timed out while waiting for an open connection"}}`. It's a bit verbose and I can change this to just match and return variant name, but I think the explicitness of this is kinda nice for internal errors.